src/components/JoinList.jsx

```diff
@@ -0,0 +1,60 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import db from '../lib/firebase';
+import { collection } from 'firebase/firestore';
+import { useCollection } from 'react-firebase-hooks/firestore';
+import { ToastContainer, toast } from 'react-toastify';
+
+export const JoinList = ({ setToken }) => {
+  const [tokenInput, setTokenInput] = useState(' ');
+
+  const notify = () => toast('Token not valid');
```

Comment:
Good call adding a toast notification for this error! Perhaps using the same styling as the success notification could be confusing for the user? I'd suggest using toast.error instead of the standard toast styling.

Reply:
+1

Reply:
Thank you! Implemented.